agent.go

```diff
@@ -200,6 +206,67 @@ func (c *container) getProcess(execID string) (*process, error) {
 	return proc, nil
 }
 
+// setSandboxStorage sets the sandbox level reference
+// counter for the sandbox storage.
+// This method also returns a boolean to let
+// callers know if the storage already existed or not.
+// It will return true if storage is new.
+//
+// It's assumed that caller is calling this method after
+// acquiring a lock on sandbox.
+func (s *sandbox) setSandboxStorage(path string) bool {
+	if _, ok := s.storages[path]; !ok {
+		sbs := &sandboxStorage{refcount: 1}
+		s.storages[path] = sbs
+		return true
+	} else {
+		sbs := s.storages[path]
+		sbs.Lock()
+		sbs.refcount++
+		sbs.Unlock()
+		return false
+	}
+}
+
+// unSetSandboxStorage will decrement the sandbox storage
+// reference counter. If there aren't any containers using
+// that sandbox storage, this method will also remove the
+// storage from the sandbox.
+//
+// It's assumed that caller is calling this method after
+// acquiring a lock on sandbox.
+func (s *sandbox) unSetSandboxStorage(path string) {
+	if sbs, ok := s.storages[path]; !ok {
+		agentLog.WithField("sandbox-storage-path", path).Error("sandbox storage not found")
+	} else {
+		sbs.Lock()
```

Comment:
You should call into `defer sbs.Unlock()` here, this way, we can add an error return without worrying about unlocking properly.